// Copyright 2011, Tencent Inc.
// Author: YU Shizhan <stanyu@tencent.com>

package poppy;

option java_package="com.soso.poppy";

enum RpcErrorCode {
    RPC_SUCCESS = 0;
    RPC_ERROR_PARSE_REQUEST_MESSAGE = 1;
    RPC_ERROR_PARSE_RESPONSE_MESSAGE = 2;
    RPC_ERROR_PARSE_RESPONES_MESSAGE = 2; // should be removed later
    RPC_ERROR_UNCOMPRESS_MESSAGE = 3;
    RPC_ERROR_COMPRESS_TYPE = 4;
    RPC_ERROR_NOT_SPECIFY_METHOD_NAME = 5;
    RPC_ERROR_METHOD_NAME = 6;
    RPC_ERROR_FOUND_SERVICE = 7;
    RPC_ERROR_FOUND_METHOD = 8;
    RPC_ERROR_CHANNEL_SHUTDOWN = 9;
    RPC_ERROR_ALL_REQUEST_DISCARDED = 9; // deprecated;
    RPC_ERROR_CONNECTION_CLOSED = 10;
    RPC_ERROR_REQUEST_TIMEOUT = 11; // server health but timeout
    RPC_ERROR_SERVER_UNAVAILABLE = 12; // server not health or health check failed
    RPC_ERROR_SERVICE_UNREACHABLE = 13; // no service process listening on address
    RPC_ERROR_SERVER_UNREACHABLE = 13;
    RPC_ERROR_NO_AUTH = 14;
    RPC_ERROR_SERVER_SHUTDOWN = 15;
    RPC_ERROR_SEND_BUFFER_FULL = 16;
    RPC_ERROR_NETWORK_UNREACHABLE = 17; // can't access to host as network reason
    RPC_ERROR_UNKNOWN = 999;
    RPC_ERROR_FROM_USER = 1000;
};
